src/js/containers/recipient/topFive/TopFiveContainer.jsx

```diff
@@ -0,0 +1,153 @@
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.recipientHash !== this.props.recipientHash) {
+            this.loadCategory();
+        }
+        if (prevProps.fy !== this.props.fy) {
+            this.loadCategory();
+        }
+    }
+
+    dataParams() {
+        let timePeriod = null;
+        if (this.props.fy === 'latest') {
+            const trailing = getTrailingTwelveMonths();
+            timePeriod = {
+                start_date: trailing[0],
+                end_date: trailing[1]
+            };
+        }
+        else if (this.props.fy !== 'all' && this.props.fy) {
+            const range = convertFYToDateRange(parseInt(this.props.fy, 10));
+            timePeriod = {
+                start_date: range[0],
+                end_date: range[1]
+            };
+        }
+
+        const filters = {
+            recipientHash: this.props.recipientHash
+        };
+
+        if (timePeriod) {
+            filters.time_period = [timePeriod];
+        }
+
+        return {
+            filters,
+            category: this.props.category,
+            limit: 5,
+            page: 1
+        };
+    }
+
+    loadCategory() {
+        if (this.request) {
+            this.request.cancel();
+        }
+
+        this.setState({
+            loading: true,
+            error: false
+        });
+
+        this.request = SearchHelper.performSpendingByCategorySearch(this.dataParams());
+        this.request.promise
+            .then((res) => {
+                this.parseResults(res.data.results, res.data.category);
+            })
+            .catch((err) => {
+                if (!isCancel(err)) {
+                    console.log(err);
+                    this.setState({
+                        loading: false,
+                        error: true
+                    });
+                }
+            });
+    }
+
+    parseResults(data, type) {
+        const parsed = data.map((item, index) => {
+            const result = Object.create(BaseStateCategoryResult);
+            result.populate(item, index + 1);
+
+            // use a special naming template for DUNS
+            if (type === 'recipient_duns') {
+                result.nameTemplate = (name) => name;
+            }
+            else if (type === 'awarding_agency' || type === 'awarding_subagency') {
+                result.nameTemplate = (name) => name;
+            }
+
+            return result;
+        });
+        this.setState({
+            loading: false,
+            error: false,
+            results: parsed
+        });
+    }
+
+    render() {
+        return (
+            <TopFive
+                category={this.props.category}
+                total={this.props.total}
+                {...this.state} />
+        );
+    }
+}
+
+
+export default connect(
+    (state) => ({
+        total: state.recipient.overview._totalAmount,
+        fy: state.recipient.fy,
+        recipientHash: state.recipient.overview.recipientHash
```

Comment:
When the recipient overview contract is updated, you will be able to access the hash via `state.recipient.id`